test/test-tcp-read-stop-start.c

```diff
@@ -0,0 +1,135 @@
+/* Copyright Joyent, Inc. and other Node contributors. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to
+ * deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+
+#include "uv.h"
+#include "task.h"
+
+static uv_tcp_t server;
+static uv_tcp_t connection;
+static int read_cb_called = 0;
+
+static uv_tcp_t client;
+static uv_connect_t connect_req;
+
+static void on_write_close_immediately(uv_write_t *req, int status) {
+  ASSERT(0 == status);
+
+  uv_close((uv_handle_t*)req->handle, NULL); /* Close immediately */
+  free(req);
+}
+
+static void on_write(uv_write_t* req, int status) {
+  ASSERT(0 == status);
+
+  free(req);
+}
+
+static void do_write(uv_stream_t *stream, uv_write_cb cb) {
+  uv_write_t *req = (uv_write_t *)malloc(sizeof(uv_write_t));
+  uv_buf_t buf;
+  buf.base = "1234578";
+  buf.len = 8;
+  ASSERT(0 == uv_write(req, stream, &buf, 1, cb));
+}
+
+static void on_alloc(uv_handle_t *handle, size_t suggested_size,
+                     uv_buf_t *buf) {
+  static char slab[65536];
+  ASSERT(suggested_size <= sizeof(slab));
+  buf->base = slab;
+  buf->len = sizeof(slab);
+}
+
+static void on_read2(uv_stream_t* stream, ssize_t nread, const uv_buf_t* buf);
+
+static void on_read1(uv_stream_t *stream, ssize_t nread, const uv_buf_t *buf) {
+  ASSERT(nread >= 0);
+
+  /* Do write on a half open connection to force WSAECONNABORTED (on Windows)
```

Comment:
I see no call to uv_shutdown, why would the connection be half-open?

Reply:
The client has closed its side of the connection.